rds-update-rdsh-collection/azuredeploy.json

```diff
@@ -0,0 +1,345 @@
+                                }
+                            }
+                        },
+                        "protectedSettings": {
+                            "Items": {
+                                "AdminPassword": "[parameters('adminPassword')]"
+                            }
+                        }
+                    }
+                },
+
+                {
+                    "type": "extensions",
+                    "name": "Microsoft.Insights.VMDiagnosticsSettings",
+                    "apiVersion": "[variables('apiVersion')]",
+                    "location": "[resourceGroup().location]",
+                    "tags": {
+                        "displayName": "AzureDiagnostics"
+                    },
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Compute/virtualMachines', concat('rdsh-', copyindex(),variables('iteration')))]"
+                    ],
+                    "properties": {
+                        "publisher": "Microsoft.Azure.Diagnostics",
+                        "type": "IaaSDiagnostics",
+                        "typeHandlerVersion": "1.5",
+                        "autoUpgradeMinorVersion": true,
+                        "settings": {
+                            "xmlCfg": "[base64(concat(variables('wadcfgx').start, 'Microsoft.Compute/virtualMachines/', 'rdsh-', copyindex(),variables('iteration'), variables('wadcfgx').end))]",
+                            "storageAccount": "[variables('storageAccountName')]"
+                        },
+                        "protectedSettings": {
+                            "storageAccountName": "[variables('storageAccountName')]",
+                            "storageAccountKey": "[listkeys(variables('accountid'), '2015-06-15').key1]",
+                            "storageAccountEndPoint": "https://core.windows.net"
+                        }
+                    }
+                }
+            ]
+        },
+
+        {
+            "apiVersion": "[variables('apiversion')]",
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "name": "cb-vm/customscript",
+            "location": "[resourceGroup().location]",
+            "dependsOn": [
+                "rdsh-vm-loop"
+            ],
+            "tags": {
+                "displayName": "script"
+            },
+            "properties": {
+                "publisher": "Microsoft.Compute",
+                "type": "CustomScriptExtension",
+                "typeHandlerVersion": "1.8",
+                "autoUpgradeMinorVersion": true,
+                "settings": {
+                    "fileUris": [
+                        "https://gallery.technet.microsoft.com/scriptcenter/Impersonate-a-User-9bfeff82/file/127189/1/New-ImpersonateUser.ps1",
+                        "https://raw.githubusercontent.com/mmarch/azure-quickstart-templates/master/rds-update-rdsh-collection/Scripts/Script.ps1"
```

Comment:
see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc for using artifacts in samples